compiler/front/in_options.nim

```diff
@@ -0,0 +1,318 @@
+    cmdJsondoc
+    cmdCtags
+    cmdBuildindex
+    cmdGendepend
+    cmdDump
+    cmdInteractive ## start interactive session
+    cmdNop
+    cmdJsonscript  ## compile a .json build file
+    cmdNimfix
+    # old unused: cmdInterpret, cmdDef: def feature (find definition for IDEs)
+
+  FilenameOption* = enum
+    ## Filename formatting option
+    foAbs           ## absolute path, e.g.: /pathto/bar/foo.nim
+    foRelProject    ## relative to project path, e.g.: ../foo.nim
+    foCanonical     ## canonical module name
+    foLegacyRelProj ## legacy, shortest of (foAbs, foRelProject)
+    foName          ## lastPathPart, e.g.: foo.nim
+    foStacktrace    ## if optExcessiveStackTrace: foAbs else: foName
+
+  Feature* = enum  ## experimental features; DO NOT RENAME THESE!
+    implicitDeref,
+    dotOperators,
+    callOperator,
+    parallel,
+    destructor,
+    notnil,
+    dynamicBindSym,
+    forLoopMacros, ## not experimental anymore; remains here for backwards
+                   ## compatibility
+    caseStmtMacros,
+    codeReordering,
+    compiletimeFFI,
+      ## This requires building nim with `-d:nimHasLibFFI`
+      ## which itself requires `nimble install libffi`, see #10150
+      ## Note: this feature can't be localized with {.push.}
+    vmopsDanger,
+    strictFuncs,
+    views,
+    strictNotNil,
+    overloadableEnums,
+    strictEffects,
+    unicodeOperators
+
+
+  LegacyFeature* = enum
+    allowSemcheckedAstModification,
+      ## Allows to modify a NimNode where the type has already been
+      ## flagged with nfSem. If you actually do this, it will cause
+      ## bugs.
+    checkUnsignedConversions
+      ## Historically and especially in version 1.0.0 of the language
+      ## conversions to unsigned numbers were checked. In 1.0.4 they
+      ## are not anymore.
+
+  TSystemCC* = enum
+    ccNone, ccGcc, ccNintendoSwitch, ccLLVM_Gcc, ccCLang, ccBcc, ccVcc,
+    ccTcc, ccEnv, ccIcl, ccIcc, ccClangCl
+
+  ExceptionSystem* = enum
+    excNone,   ## no exception system selected yet
+    excSetjmp, ## setjmp based exception handling
+    excCpp,    ## use C++'s native exception handling
+    excGoto,   ## exception handling based on goto (should become the new default for C)
+    excQuirky  ## quirky exception handling
+
+  SymbolFilesOption* = enum
+    disabledSf   ## disables Rod files and maybe packed AST features
+    writeOnlySf  ## not really sure, beyond not reading rod files
+    readOnlySf   ## we only read from rod files
+    v2Sf         ## who knows, probably a bad idea
+    stressTest   ## likely more bad ideas
+
+type
+  ConfNoteSet* = enum
+    cnCurrent ## notes after resolving all logic(defaults,
+              ## verbosity)/cmdline/configs
+    cnMainPackage
+    cnForeign
+    cnWarnAsError
+    cnHintAsError
+    cnCmdline ## notes that have been set/unset from cmdline
+    cnModifiedy ## notes that have been set/unset from either
+                ## cmdline/configs
+
+type
+  ActiveConf* = object
```

Comment:
```suggestion
  CurrentConf* = object
```

I suggest renaming this to `CurrentConf` as it's the current understanding of the compiler configuration/input.

As a stretch goal rename ConfigRef to `CompilerStateRef`, because it's a hilarious global -- or we can tackle that separately.